frontend/src/lib/utils.tsx

```diff
@@ -632,12 +632,29 @@ export function dateFilterToText(
     }
     dateFrom = (dateFrom || undefined) as string | undefined
     dateTo = (dateTo || undefined) as string | undefined
+
     if (isDate.test(dateFrom || '') && isDate.test(dateTo || '')) {
         return `${dateFrom} - ${dateTo}`
     }
+
     if (dateFrom === 'dStart') {
+        // Changed to "last 24 hours" but this is backwards compatibility
         return 'Today'
-    } // Changed to "last 24 hours" but this is backwards compatibility
+    }
+
+    if (isDate.test(dateFrom || '') && !isDate.test(dateTo || '')) {
+        const days = dayjs().diff(dayjs(dateFrom), 'days')
+        if (days > 366) {
+            return `${dateFrom} - Today`
+        } else if (days > 0) {
+            return `Last ${days} days`
+        } else if (days === 0) {
+            return `Today`
+        } else {
+            return `Starting from ${dateFrom}`
```

Comment:
This condition would never be met right?

Reply:
Who knows :)

Anyone can edit the URL, set the from date to something in the future and then get to this.